ui/appui-react/src/appui-react/settings/ui/UiSettingsPage.tsx

```diff
@@ -33,7 +33,20 @@ import { Select, SelectOption, Slider, ToggleSwitch } from "@itwin/itwinui-react
  *
  * @beta
  */
-export function UiSettingsPage({ allowSettingUiFrameworkVersion }: { allowSettingUiFrameworkVersion: boolean }) {
+export function UiSettingsPage(): JSX.Element;
+
+/**
+ * @deprecated Framework version is deprecated, only UI2.0 is supported.
+ * @beta
+ */
+export function UiSettingsPage({ allowSettingUiFrameworkVersion }: { allowSettingUiFrameworkVersion: boolean }): JSX.Element; // eslint-disable-line @typescript-eslint/unified-signatures
+
+/**
+ * @deprecated
+ * @beta
+ */
+export function UiSettingsPage(props?: { allowSettingUiFrameworkVersion: boolean }) {
```

Comment:
Do you need both?

Reply:
Yes, you need to define a declaration for each overload and a shared implementation.

Reply:
And the implementation also need the tags ?

Reply:
Not necessarily, since only declarations are visible in the definition files, but linter complains if there is no comment at the top.